src/lib/wallet/SoftwareWalletProvider.js

```diff
@@ -84,7 +84,10 @@ class SoftwareWalletProvider {
     if (web3) {
       // https://github.com/ChainSafe/web3.js/issues/4780
       // use web3.eth.personal.sign as workaround for metamask
-      const sign = web3.currentProvider?.isMetaMask ? web3.eth.personal.sign : web3.eth.sign
+      const sign =
+        web3.currentProvider?.isMetaMask || web3.currentProvider.walletMeta?.name?.toLowerCase() === 'metamask'
+          ? web3.eth.personal.sign
+          : web3.eth.sign
```

Comment:
can we find a more generic condition? is this metamask related only?

Reply:
This workaround is specific for metamask only.